packages/licenses/src/getPkgInfo.ts

```diff
@@ -0,0 +1,356 @@
+      return {
+        local: false,
+        files,
+      }
+    } catch (err: any) {  // eslint-disable-line
+      if (err.code === 'ENOENT') {
+        throw new PnpmError(
+          'MISSING_PACKAGE_INDEX_FILE',
+          `Failed to find package index file for ${packageRef}, please consider running 'pnpm install'`
+        )
+      }
+
+      throw err
+    }
+  }
+}
+
+/**
+ * Returns the package manifest information for a give package name and path
+ * @param pkg the package details to lookup info for
+ */
+export const getPkgInfo: GetPackageInfoFunction = async (
+  pkg,
+  opts
+): Promise<
+{
+  from: string
+  description?: string
+} & Omit<LicensePackage, 'belongsTo'>
+> => {
+  const cafsDir = path.join(opts.storeDir, 'files')
+
+  // Retrieve file index for the requested package
+  const packageResolution = pkgSnapshotToResolution(
+    pkg.depPath,
+    pkg.snapshot,
+    pkg.registries
+  )
+
+  const packageFileIndexInfo = await readPackageIndexFile(
+    packageResolution as Resolution,
+    pkg.depPath,
+    {
+      cafsDir,
+      storeDir: opts.storeDir,
+      lockfileDir: opts.dir,
+    }
+  )
+
+  // Fetch the package manifest
+  let packageManifestDir: string = ''
+  if (packageFileIndexInfo.local) {
+    packageManifestDir = packageFileIndexInfo.files['package.json']
+  } else {
+    const packageFileIndex = packageFileIndexInfo.files as Record<
+    string,
+    PackageFileInfo
+    >
+    const packageManifestFile = packageFileIndex['package.json']
+    packageManifestDir = await getFilePathByModeInCafs(
+      cafsDir,
+      packageManifestFile.integrity,
+      packageManifestFile.mode
+    )
+  }
+
+  let manifest
+  try {
+    manifest = await readPkg(packageManifestDir)
+  } catch (err: any) {  // eslint-disable-line
+    if (err.code === 'ENOENT') {
+      throw new PnpmError(
+        'MISSING_PACKAGE_MANIFEST',
+        'Failed to find package manifest file'
+      )
+    }
+    throw err
+  }
+
+  // Determine the path to the package as known by the user
+  const modulesDir = opts.modulesDir ?? 'node_modules'
+  const virtualStoreDir = pathAbsolute(
+    opts.virtualStoreDir ?? path.join(modulesDir, '.pnpm'),
+    opts.dir
+  )
+
+  const packageModulePath = path.join(
+    virtualStoreDir,
+    depPathToFilename(pkg.depPath),
+    modulesDir,
+    manifest.name
+  )
+
+  const licenseInfo = await parseLicense(
+    { manifest, files: packageFileIndexInfo },
+    { cafsDir }
+  )
+
+  const packageInfo = {
+    from: manifest.name,
+    path: packageModulePath,
+    name: manifest.name,
+    version: manifest.version,
+    description: manifest.description,
+    license: licenseInfo.name,
+    licenseContents: licenseInfo.licenseFile,
+    vendorName:
+      (manifest.author &&
+        (typeof manifest.author === 'string'
+          ? manifest.author
+          : (manifest.author as { name: string }).name)) ??
+      undefined,
+    vendorUrl: manifest.homepage,
```

Comment:
Use the existing terminology: author and homepage.

Reply:
Updated the code to use `author` and `homepage` everywhere